core/src/main/java/org/testcontainers/images/LocalImagesCache.java

```diff
@@ -79,7 +79,11 @@ private void populateFromList(List<Image> images) {
             }
 
             cache.putAll(
-                Stream.of(repoTags).collect(Collectors.toMap(
+                Stream.of(repoTags)
+                    // Protection against some edge case where local image repository tags end up with duplicates
+                    // making toMap crash at merge time.
+                   .distinct()
```

Comment:
alternatively, we one can provide a 3rd argument to `toMap` that won't throw

Reply:
Merge function like (u, u2) -> u ?

Would do the same thing with slightly worse performance  (merge function would be applied after key and value computation), no ? 

Up to you, tell me what you prefer.

Reply:
Ok, you're right, let's keep `distinct()` 👍